metagraph/src/graph/representation/succinct/boss.cpp

```diff
@@ -639,6 +639,27 @@ edge_index BOSS::fwd(edge_index i, TAlphabet c) const {
     return select_last(target_node);
 }
 
+/**
+ * This function gets a position i reflecting the r-th occurrence of the corresponding
+ * character c in W and returns the position of the r-th occurrence of c in last.
+ */
+edge_index BOSS::fwd(edge_index i) const {
+    CHECK_INDEX(i);
+
+    auto [rank, c] = W_->inverse_select(i);
+    assert(i == 1 || c != kSentinelCode);
+    if (c == kSentinelCode) {
+        --rank;
+    } else if (c >= alph_size) {
+        c %= alph_size;
+        rank = rank_W(i, c);
+    }
```

Comment:
`inverse_select` + `rank` might actually be more expensive than access + rank. Did you benchmark this for all types?